src/test/java/com/google/api/generator/engine/ast/RelationalOperationExprTest.java

```diff
@@ -0,0 +1,431 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.engine.ast;
+
+import static org.junit.Assert.assertThrows;
+
+import com.google.api.generator.engine.ast.TypeNode.TypeKind;
+import org.junit.Test;
+
+public class RelationalOperationExprTest {
```

Comment:
This is nice and thorough! Some more scenarios to consider:
- Numeric object and some Object
- For each test, flip the RHS and LHS values (in the same test) so we can ensure we're not just looking at the LHS types.
- Any type and null.
- null and null.

Reply:
Thanks your suggestions. I added more test cases.
- Numeric object and some Object (See LHS Numeric Boxed type, compare with Object, String ) [code](https://github.com/googleapis/gapic-generator-java/pull/289/files#diff-eb49f476bfd2402d83757f2bffd6bb6cR132-R219) 
- For each test, flip the RHS and LHS values (in the same test) so we can ensure we're not just looking at the LHS types. (Added missed flip test case and comment in each test)
- Any type and null. (Added in each type)
- null and null (add the [test](https://github.com/googleapis/gapic-generator-java/pull/289/files#diff-eb49f476bfd2402d83757f2bffd6bb6cR632-R639)).

Reply:
Thanks for the detailed test cases.